src/mmap_storage.cpp

```diff
@@ -164,6 +164,7 @@ error_code translate_error(std::error_code const& err, bool const write)
 			if (old_prio == dont_download && new_prio != dont_download)
 			{
 				// move stuff out of the part file
+				m_file_priority[i] = new_prio;
```

Comment:
I take it this needs to be set early, so that when files are created, the priority is non-zero, to trigger the file allocation.
That should be a comment here, otherwise someone reading this function in the future may think this is redundant and remove it. (unless there's a test covering this).

Reply:
Sure.

Reply:
But now when I'm thinking about it, maybe it's better to move setting the priority up, instead of duplicating it?

I.e.
```patch
--- a/src/mmap_storage.cpp
+++ b/src/mmap_storage.cpp
@@ -161,6 +161,8 @@ error_code translate_error(std::error_code const& err, bool const write)
 
 			download_priority_t const old_prio = m_file_priority[i];
 			download_priority_t new_prio = prio[i];
+			m_file_priority[i] = new_prio;
+
 			if (old_prio == dont_download && new_prio != dont_download)
 			{
 				// move stuff out of the part file
@@ -266,7 +268,6 @@ error_code translate_error(std::error_code const& err, bool const write)
 */
 			}
 			ec.ec.clear();
-			m_file_priority[i] = new_prio;
 
 			if (m_file_priority[i] == dont_download && use_partfile(i))
 			{

```

This way looks cleaner. Idk why it was at the bottom in the first place, but behavior shouldn't change, as far as I can see.

Reply:
Ok, I pushed the new iteration, should make it better overall.